format/spec.md

```diff
@@ -1400,7 +1400,7 @@ In some cases partition specs are stored using only the field list instead of th
 
 The `field-id` property was added for each partition field in v2. In v1, the reference implementation assigned field ids sequentially in each spec starting at 1,000. See Partition Evolution for more details.
 
-In v3 metadata, writers must use only `source-ids` because v3 requires reader support for multi-arg transforms. In v1 and v2 metadata, writers must always write `source-id`; for multi-arg transforms, writers must produce `source-ids` and set `source-id` to the first ID from the field ID list.
+In v3 metadata, writers must use only `source-ids` because v3 requires reader support for multi-arg transforms.
```

Comment:
That's ok but I guess we have to clarify the expectations for the readers. Afair, for the readers, the spec says that if both `source-id` and `source-ids` are provided (in the case of an upgrade from V1/V2 to V3), then `source-id` should be used for single arg transforms and `source-ids` for multi arg transforms. I assume that if `source-id` and `source-ids` are provided in V3 (coming from V1/V2), single arg transforms should use the first element of `source-ids` and ignore `source-id`. I think we should clear the strategy here for the readers as well.

Reply:
You remember that correctly :) This PR doesn't touch the read-section:

https://github.com/apache/iceberg/blob/f6faa58dac57e03be6e02a43937ac7c15c770225/format/spec.md?plain=1#L1562-L1565

For V3 the `source-id` should be omitted:

https://github.com/apache/iceberg/blob/f6faa58dac57e03be6e02a43937ac7c15c770225/format/spec.md?plain=1#L1381

Let me know if you feel like more clarification is needed.